lib/link_checker.rb

```diff
@@ -0,0 +1,49 @@
+require "uri"
+
+module LinkChecker
```

Comment:
This really feels like it should be a class rather than a module. Using a module means that the Rake task using this functionality has to know a lot about how it works - the logic is split between the rake task and the module, because the module methods are effective chunks of functional code.

I think it would be better to initiate an instance, passing in a hash of strings (output from a `YAML.safe_load`) on initialization, and then output a report.